### ISSUE TITLE
能给出一个完整的onnxruntime推理demo吗

### ISSUE THREAD

**Issue by Zheng-Bicheng** [meituan/YOLOv6#272]
能不能给出一个完整的python或者c++怎么调用onnxruntime来进行推理，真不是伸手党。看了一下推理的过程，letterbox后的图片大小从来就不一致。根据不同的图片，letterbox会产生不同大小的结果，而不是固定的640*640。但是，onnx模型的输入不是固定的吗，这样让我们怎么输入呢?

**Comment by zhiqwang** [comment]
ONNX Runtime支持动态shape推理

**Comment by Zheng-Bicheng** [comment]
before letter box,image shape = (536, 640, 3)
after letter box,image shape = (544, 640, 3)
before letter box,image shape = (480, 640, 3)
after letter box,image shape = (480, 640, 3)
before letter box,image shape = (640, 431, 3)
after letter box,image shape = (640, 448, 3)

**Comment by LYHTZHANGJIANG** [comment]
想问一下，你量化编译到RK上面之后数据置信度都是0了,请问你是正常的吗？

**Comment by scutzhe** [comment]
我使用onnx推理出来的错位了，还在debug中
![2022-07-19-14:51:04](https://user-images.githubusercontent.com/18154839/179685243-125daec5-93ca-4e0c-86c7-12f33852ffdf.png)

**Comment by scutzhe** [comment]
`import cv2
import torch
import numpy as np
import onnxruntime


def letterbox(im, new_shape=(640, 640), color=(114, 114, 114), auto=False, scaleup=True, stride=32):
    """
    @param im:
    @param new_shape:
    @param color:
    @param auto:
    @param scaleup:
    @param stride:
    @return:
    """
    shape = im.shape[:2]  # current shape [height, width]
    if isinstance(new_shape, int):
        new_shape = (new_shape, new_shape)

    # Scale ratio (new / old)
    r = min(new_shape[0] / shape[0], new_shape[1] / shape[1])
    if not scaleup:  # only scale down, do not scale up (for better val mAP)
        r = min(r, 1.0)

    # Compute padding
    new_unpad = int(round(shape[1] * r)), int(round(shape[0] * r))
    dw, dh = new_shape[1] - new_unpad[0], new_shape[0] - new_unpad[1]  # wh padding

    if auto:  # minimum rectangle
        dw, dh = np.mod(dw, stride), np.mod(dh, stride)  # wh padding

    dw /= 2  # divide padding into 2 sides
    dh /= 2

    if shape[::-1] != new_unpad:  # resize
        im = cv2.resize(im, new_unpad, interpolation=cv2.INTER_LINEAR)
    top, bottom = int(round(dh - 0.1)), int(round(dh + 0.1))
    left, right = int(round(dw - 0.1)), int(round(dw + 0.1))
    im = cv2.copyMakeBorder(im, top, bottom, left, right, cv2.BORDER_CONSTANT, value=color)  # add border
    return im, r, (dw, dh)


class OnnxInference(object):
    def __init__(self,onnx_path="pth/head_200_yuncong.onnx"):
        self.session = onnxruntime.InferenceSession(onnx_path, providers=['TensorrtExecutionProvider',
                                                                     'CUDAExecutionProvider',
                                                                     'CPUExecutionProvider'])
    def __call__(self,image):
        """
        np's format pic
        @param image:
        @return:
        """
        image = self.precess_image(image)
        input_name = self.session.get_inputs()[0].name
        output_name = self.session.get_outputs()[0].name
        output = self.session.run([output_name], {input_name: image})
        return output


    @staticmethod
    def precess_image(img_src, img_size=640, stride=32, half=False):
        """
        @param img_src:
        @param img_size:
        @param stride:
        @param half:
        @return:
        """
        try:
            image = letterbox(img_src, img_size, stride=stride)[0]
            image = image.transpose((2, 0, 1))[::-1]  # HWC to CHW, BGR to RGB
            image = torch.from_numpy(np.ascontiguousarray(image))
            image = image.unsqueeze(0)
            image = image.half() if half else image.float()  # uint8 to fp16/32
            image /= 255  # 0 - 255 to 0.0 - 1.0
            return np.array(image)

        except Exception as e:
            print("image predeal {}".format(e))`
